pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -110,10 +113,12 @@ func NewController(client kube.Client, meshWatcher mesh.Holder,
 		classes:      classes,
 		services:     services,
 	}
-	c.queue = controllers.NewQueue("ingress",
-		controllers.WithReconciler(c.onEvent),
-		controllers.WithMaxAttempts(5))
-	c.ingress.AddEventHandler(controllers.ObjectHandler(c.queue.AddObject))
+	c.queue = queue.NewQueueWithID(time.Second, "ingress")
+	c.ingress.AddEventHandler(controllers.FromEventHandler(func(o controllers.Event) {
```

Comment:
This seems like a regression. We have been intentionally moving controllers _to_ the `controllers.Queue` from the `queue.Instance`, to more closely align with Kubernetes best practice on controller reconciliation.

Reply:
Thankfully, ingress controller stores old ingress, so i think can restore to `controllers.Queue`, and rely on the comparition of the new/old ingress